include/proc.h

```diff
@@ -24,6 +25,8 @@ struct proc {
   vm_map_t *p_uspace;      /* process' user space map */
   /* file descriptors table */
   fdtab_t *p_fdtable;
+  /* signal stuff */
+  sighand_t *p_sighand; /* Signal handlers data */
```

Comment:
I see no reason for introduction `sighand_t` structure. You could just replace it with:
```c
sigaction_t p_sigacts[SIG_LAST];
```
... and remove some boilerplate code. We won't share signal handlers between processes in any foreseeable future. Frankly, I don't care about `rfork` nor `clone` implementation.